src/screens/DiscoveryScreenV2/elements/DiscoveryTitleSeparator.js

```diff
@@ -1,35 +1,78 @@
 import * as React from 'react';
-import {Text} from 'react-native';
-
+import {Animated, Text, View} from 'react-native';
+import ArrowDown from '../../../../assets/icons/arrow-down.svg';
 import {fonts} from '../../../utils/fonts';
 import {COLORS} from '../../../utils/theme';
 
 /**
  * @typedef {Object} DiscoveryTitleSeparatorProp
  * @property {String} text
+ * @property {Boolean} showArrow
+ * @property {Boolean} [rotateArrow]
  */
 /**
  *
- * @param {DiscoveryTitleSeparatorProp} prop
+ * @param {DiscoveryTitleSeparatorProp} props
  */
-const DiscoveryTitleSeparator = (prop) => {
-  const {text} = prop;
+const DiscoveryTitleSeparator = (props) => {
+  const {text, showArrow, rotateArrow = false} = props;
+
+  // Animation value for rotation
+  const rotation = React.useRef(new Animated.Value(0)).current;
+
+  // Interpolating the animation value to create a rotation style
+  const rotate = rotation.interpolate({
+    inputRange: [0, 1],
+    outputRange: ['0deg', '180deg'] // Rotate 180 degrees
+  });
+
+  // Trigger rotation animation if `rotateArrow` is true
+  React.useEffect(() => {
+    if (rotateArrow) {
+      Animated.timing(rotation, {
+        toValue: 1,
+        duration: 300, // Adjust as needed
+        useNativeDriver: true
+      }).start();
+    } else {
+      // Reset to default rotation
+      Animated.timing(rotation, {
+        toValue: 0,
+        duration: 300,
+        useNativeDriver: true
+      }).start();
+    }
+  }, [rotateArrow]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The animation logic for rotating the arrow is well implemented using React hooks and Animated API from React Native. However, it's important to stop any ongoing animations before starting a new one to prevent unexpected behavior. You can achieve this by calling `stopAnimation` before `start`.

```diff
  React.useEffect(() => {
    rotation.stopAnimation(); // Stop any ongoing animations
    if (rotateArrow) {
      Animated.timing(rotation, {
        toValue: 1,
        duration: 300, // Adjust as needed
        useNativeDriver: true
      }).start();
    } else {
      // Reset to default rotation
      Animated.timing(rotation, {
        toValue: 0,
        duration: 300,
        useNativeDriver: true
      }).start();
    }
  }, [rotateArrow]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->